src/api.v2/controllers/sampleController.js

```diff
@@ -52,7 +52,7 @@ const deleteSample = async (req, res) => {
   logger.log(`Deleting sample ${sampleId} from experiment ${experimentId}`);
 
   await sqlClient.get().transaction(async (trx) => {
-    await new Sample(trx).destroy(sampleId);
+    await new Sample(trx).delete(sampleId);
```

Comment:
Will the metadata track entries, if any, be deleted when we call `Sample.delete` in line 55?

Reply:
I didn't know this about v1. How do `sample_in_metadata_track_map` get auto deleted? Is there a trigger?

Reply:
it's the result of th eforeign keys set up.
If you write ON DELETETE CASCADE you're saying that when the foreign key a row is referencing is deleted then its own action should be to also be deleted.